src/conffile.c

```diff
@@ -621,13 +622,17 @@ static int get_fqdn(struct conf **c)
 		goto end;
 	}
 
-	if(set_string(c[OPT_CNAME], info->ai_canonname))
+	fqdn=strdup(info->ai_canonname);
```

Comment:
Needs to be strdup_w, and check that the result is not NULL:
if(!(fqdn=strdup_w(info->ai_canonname, __func__)))
  goto end;